auto_nag/scripts/severity_high_compat_priority.py

```diff
@@ -0,0 +1,57 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+
+# TODO: should be moved when resolving https://github.com/mozilla/relman-auto-nag/issues/1384
+LOW_SEVERITY = ["S3", "normal", "S4", "minor", "trivial", "enhancement"]
+
+
+class SeverityHighCompatPriority(BzCleaner):
+    def description(self):
+        return "Bugs with P1 WebCompat priority and severity set to low"
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        data[bugid] = {
+            "severity": bug["severity"],
+        }
+        self.extra_ni = data
+
+        return bug
+
+    def get_extra_for_needinfo_template(self):
+        return self.extra_ni
+
+    def columns(self):
+        return ["id", "summary", "severity"]
+
+    def get_mail_to_auto_ni(self, bug):
+        for field in ["assigned_to", "triage_owner"]:
+            person = bug.get(field, "")
+            if person and not utils.is_no_assignee(person):
+                return {"mail": person, "nickname": bug[f"{field}_detail"]["nick"]}
+
+        return None
+
+    def get_bz_params(self, date):
+        fields = ["triage_owner", "assigned_to", "severity"]
+
+        params = {
+            "include_fields": fields,
+            "resolution": "---",
```

Comment:
Let's only do this for bugs of type `defect`, as severity doesn't apply to other types.

Reply:
Make sense. Done in 27c1cf2410aad911cd5f3fece345eb7d418875d4




<details>
<summary>Click to expand the dry-run output!</summary>

<br>

<p>The following bugs have a P1 WebCompat priority, however, their severity is set to low. Owners got needinfo requests to bump the severity:
<table style="border:1px solid black;border-collapse:collapse;" border="1">
<thead>
<tr>
<th>Bug</th><th>Summary</th><th>Severity</th>
</tr>
</thead>
<tbody>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1752655">1752655</a>
</td>
<td>
Slack issues on session restore: custom font not loaded, sidebar with thread conversation broken or: &#34;For some reason, Slack couldn&#8217;t load&#34;
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1680999">1680999</a>
</td>
<td>
www.bose.com - Instant  browser crash
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1651588">1651588</a>
</td>
<td>
Text deleted in contenteditable=&#34;true&#34; element on drag and drop
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1615852">1615852</a>
</td>
<td>
broken `&lt;br&gt;` workaround in `&lt;div contenteditable=&#34;true&#34;&gt;` with bad side effects (also affects Google Sheets)
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1524772">1524772</a>
</td>
<td>
Webcompat: Offline unavailable in Gmail
</td>
<td>
normal
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1520297">1520297</a>
</td>
<td>
Preload Intermediates on Android
</td>
<td>
normal
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1423746">1423746</a>
</td>
<td>
CSS &#39;filter&#39; on the root element (documentElement) should not establish a containing block for absolute/fixed positioned descendants
</td>
<td>
normal
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1123938">1123938</a>
</td>
<td>
[meta] Make viewport handling consistent with other browsers
</td>
<td>
normal
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=92921">92921</a>
</td>
<td>
contenteditable should not replace linebreaks by &lt;br&gt; inside a &lt;pre&gt;
</td>
<td>
S3
</td>
</tr>
</tbody>
</table>
</p>
<p>Sincerely,<br>
Autonag bot
</p>
<p>
See the search query on <a href="https://mzl.la/3ytRuB8">Bugzilla</a>.
</p>

<p>
See the source / report issues on the <a href="https://github.com/mozilla/relman-auto-nag/">GitHub repo</a>
</p>

</details>